Providers/PythonProvider.cpp

```diff
@@ -48,22 +49,51 @@ char const SCRIPT_PATH_EXTENSION[] = "/lib/Scripts/";
 char const DEFAULT_DSC_SCRIPT[] = "client";
 char const PY_EXTENSION[] = ".py";
 
+std::string determinePythonVersion(){
+    std::array<char, 128> buffer;
+    std::string result;
+    // Check for python2
+    FILE* pipe = popen("python2 --version 2>&1", "r");
+    if(!pipe) {
+        std::cout << "Couldn't start command." << std::endl;
+    }
+
+    while(fgets(buffer.data(), 128, pipe) != NULL) {
+        result += buffer.data();
+    }
+
+    // If python2 --version does not contain 'not found' return python2
+    if(result.find("not found") == std::string::npos) {
+      std::cout << "Found python2." << std::endl;
+    	return "python";
+    }
+
+    // Look for python3
+    result = "";
+    pipe = popen("python3 --version 2>&1", "r");
+    if(!pipe) {
+    	std::cout << "Couldn't start command." << std::endl;
+    }
+    while(fgets(buffer.data(), 128, pipe) != NULL) {
+      result += buffer.data();
+    }
+
+    // If python3 --version does not contain 'not found' return python3
+    if(result.find("not found") == std::string::npos) {
+      std::cout << "Found python3." << std::endl;
+	    return "python3";
+    }
+    return "python";
+}
 
 char_array::move_type
 get_python_version ()
 {
-    char* sPath = getenv (OMI_PYTHON_VERSION_STR);
     char_array pyV;
-    if (sPath == NULL)
-    {
-        pyV.reset (strcpy (new char[1 + strlen (DEFAULT_PYTHON_VERSION)],
-                           DEFAULT_PYTHON_VERSION));
-    }
-    else
-    {
-        pyV.reset (strcpy (new char[1 + strlen (sPath)], sPath));
-    }
-    return pyV.move ();
+    std::cout << "In PythonProvider." << std::endl;
+    std::string version = determinePythonVersion();
+    pyV.reset (strcpy (new char[1 + version.length()], version.c_str()));
+    return pyV.move();
```

Comment:
Please check the path for client.py when running on python3 version. It throws up an error saying file not found.

Reply:
In Client.py, When the python runtime is 3.x. There is some problem with the relative imports. I think an explicit add of the scripts is needed and then import Scripts. This way it adds all the resources scripts for it to execute. Contact me if you need further information on this
sys.path.append(ScriptsDir + '/3.x/Scripts')
from Scripts import *